src/Mvc/test/Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.Test/DependencyInjection/RazorRuntimeCompilationMvcCoreBuilderExtensionsTest.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.Razor.Compilation;
+using Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation;
+using Xunit;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public class RazorRuntimeCompilationMvcCoreBuilderExtensionsTest
+    {
+        [Fact]
+        public void AddServices_ReplacesRazorViewCompiler()
+        {
+            // Arrange
+            var services = new ServiceCollection()
+                .AddSingleton<IViewCompilerProvider, DefaultViewCompilerProvider>();
+
+            // Act
+            RazorRuntimeCompilationMvcCoreBuilderExtensions.AddServices(services);
+
+            // Assert
+            var serviceDescriptor = Assert.Single(services, service => service.ServiceType == typeof(IViewCompilerProvider));
+            Assert.Equal(typeof(RuntimeViewCompilerProvider), serviceDescriptor.ImplementationType);
```

Comment:
Minor but `Assert.IsType<RuntimeViewCompilerProvider>(...)` is easier to read

Reply:
That only works if you're testing an instance e.g. `Assert.IsType<RuntimeViewCompilerProvider>(value);`, not when you're just asserting two instances of `Type`